src/app/components/article-dialog/article-dialog.component.html

```diff
@@ -0,0 +1,40 @@
+<h2>Adding a new article</h2>
+
+<form class="tab-pane-form" [formGroup]="form">
+  <mat-form-field hintLabel="Min 2 characters" appearance="fill">
+    <mat-label>Enter author</mat-label>
+    <input matInput #author formControlName="author">
+    <mat-hint align="end">{{author.value?.length || 0}}</mat-hint>
+    <mat-error *ngIf="form.get('author')?.invalid && form.get('author')?.touched">
+      <small *ngIf="form.get('author')?.errors?.['required']">Required field</small>
+      <small *ngIf="form.get('author')?.errors?.['minlength']">
+        Min length should be {{ form.get("author")?.errors?.['minlength']['requiredLength'] }} numbers
```

Comment:
let's use a map like ErrorMap<ErrorKey, ErrorDescription> and inside class getErrorMessage for control from ErrorMap